core/src/main/kotlin/com/karumi/kotlinsnapshot/core/SerializationModule.kt

```diff
@@ -0,0 +1,65 @@
+package com.karumi.kotlinsnapshot.core
+
+import kotlin.reflect.KProperty1
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.memberProperties
+
+interface SerializationModule<in A> {
+
+    fun serialize(value: A?): String
+}
+
+class KotlinSerialization : SerializationModule<Any> {
+
+    override fun serialize(value: Any?): String = when {
+        value == null -> "null"
+        isPrimitive(value) -> value.toString()
+        value is CharSequence -> value.toString()
+        value::class.java.isEnum -> value.toString()
+        value is Iterable<*> -> iterableToString(value)
+        value is Array<*> -> iterableToString(value.toList())
+        value is Map<*, *> -> mapToString(value)
+        value is Pair<*, *> -> pairToString(value)
+        else -> toString(value)
+    }
+
+    private fun isPrimitive(value: Any) =
+        value::class.javaPrimitiveType?.isPrimitive ?: false
+
+    private fun toString(value: Any): String {
+        val anyClass = value::class
+        val className = anyClass.simpleName
+        val fieldValueList = anyClass.memberProperties.joinToString { field ->
```

Comment:
We should sort them by name so that the implementation of the inner `memberProperties` collection doesn't affect the outcome

Reply:
Yes, I'm going to sort and test it

Reply:
Not sure how are we going to test it if the kotlin team doesn't change the implementation.